internals/src/array_vec.rs

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! A simplified `Copy` version of `arrayvec::ArrayVec`.
+
+use core::fmt;
+
+pub use safety_boundary::ArrayVec;
+
+/// Limits the scope of `unsafe` auditing.
+mod safety_boundary {
+    use core::mem::MaybeUninit;
+    use crate::const_tools::cond_const;
+
+    /// A growable contiguous collection backed by array.
+    #[derive(Copy)]
+    pub struct ArrayVec<T: Copy, const CAP: usize> {
+        len: usize,
+        data: [MaybeUninit<T>; CAP],
+    }
+
+    impl<T: Copy, const CAP: usize> ArrayVec<T, CAP> {
+        // bounds are unstable until 1.61
+        cond_const! {
```

Comment:
Is this name `cond_` a convention? I didn't catch the meaning until I read the maco, perhaps `maybe_const!`?

Reply:
IDK, I made it up from "conditional compilation". Maybe `cfg_const` would be nicer.

Reply:
No worries. I prefer `cond_const!` over `cfg_const!` (but FWIW I prefer `maybe_cons!` more still). I'm ok with any of them though.

Reply:
Let's just not create more work :)

Reply:
I think `cond_` is perfectly clear. `cons` I read as in "cons cell" which is the building block of a Lisp list.

If we want to rename it I we could extend it to `conditional_const` but I really don't care and agree "let's not create more work" :).